analyzers/src/SonarAnalyzer.Common/Rules/CommentsShouldNotBeEmptyBase.cs

```diff
@@ -0,0 +1,144 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class CommentsShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4663";
+
+    protected abstract string GetCommentText(SyntaxTrivia trivia);
+
+    protected override string MessageFormat => "Remove this empty comment";
+
+    protected CommentsShouldNotBeEmptyBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterTreeAction(Language.GeneratedCodeRecognizer, c =>
+        {
+            foreach (var token in c.Tree.GetRoot().DescendantTokens())
+            {
+                CheckTrivia(c, token.LeadingTrivia);
+                CheckTrivia(c, token.TrailingTrivia);
+            }
+        });
+
+    private void CheckTrivia(SonarSyntaxTreeReportingContext context, IEnumerable<SyntaxTrivia> trivia)
+    {
+        var partitions = PartitionComments(trivia);
+        if (partitions is null)
+        {
+            return;
+        }
+
+        foreach (var partition in partitions.Where(trivia => trivia.Any() && trivia.All(x => string.IsNullOrWhiteSpace(GetCommentText(x)))))
+        {
+            var start = partition.First().GetLocation().SourceSpan.Start;
+            var end = partition.Last().GetLocation().SourceSpan.End;
+            var location = Location.Create(context.Tree, TextSpan.FromBounds(start, end));
+            context.ReportIssue(Diagnostic.Create(Rule, location));
+        }
+    }
+
+    private List<List<SyntaxTrivia>> PartitionComments(IEnumerable<SyntaxTrivia> trivia)
+    {
+        // Hotpath: avoid unnecessary allocations
+        List<List<SyntaxTrivia>> partitions = null;
+        List<SyntaxTrivia> current = null;
+        var firstEndOfLineFound = false;
+
+        foreach (var trivium in trivia)
+        {
+            if (IsSimpleComment(trivium)) // put it on the current block of "//"
+            {
+                AddTriviaToPartition(ref current, trivium);
+                firstEndOfLineFound = false;
+            }
+            // This is for the case, of two different comment types, for example:
+            // //
+            // ///
+            else if (IsValidTriviaType(trivium)) // valid but not "//", because of the upper if
+            {
+                CloseCurrentPartition(ref current, ref partitions, ref firstEndOfLineFound);
+                // all comments except single-line comments are parsed as a block already.
+                AddTriviaToPartition(ref current, trivium);
+                CloseCurrentPartition(ref current, ref partitions, ref firstEndOfLineFound);
+            }
+            // This handles an empty line, for example:
+            // // some comment \n <- EOL found, set to true
```

Comment:
firstEndOfLineFound should be mentioned once to make clear what the "true/false" refers to.
```suggestion
            // // some comment \n <- EOL found, firstEndOfLineFound set to true
```